src/AppBundle/Controller/Admin/Client/ReportController.php

```diff
@@ -162,15 +162,17 @@ public function checklistAction(Request $request, $id)
             $checklist->setButtonClicked($buttonClicked->getName());
         }
         if ($form->isValid($buttonClicked)) {
+            $noticeAlreadyThere = $request->getSession()->getFlashBag()->has('notice');
             if (!empty($checklist->getId())) {
                 $this->getRestClient()->put('report/' . $report->getId() . '/checked', $checklist, [
                     'report-checklist', 'checklist-information'
                 ]);
-                $request->getSession()->getFlashBag()->add('notice', 'Lodging checklist saved');
             } else {
                 $this->getRestClient()->post('report/' . $report->getId() . '/checked', $checklist, [
                     'report-checklist', 'checklist-information'
                 ]);
+            }
+            if(!$noticeAlreadyThere) {
```

Comment:
To avoid confusion: consider putting your if condition directly inline with an explanation comment:
if (!$request->getSession()->getFlashBag()->has('notice') {
   // the duplicate notice is because the PDF view action doesn't actually refresh the page and therefor the original 
   // 'saved' notice never gets rendered